### ISSUE TITLE
Don't call `object_class` on serializers in validate_presence_of

### ISSUE THREAD

**Issue by theycallmeswift** [thoughtbot/shoulda-matchers#1235]
Hey folks --

I'm using a custom module for serialization and have a validator on presence. There's no guarantee that a module responds to the `object_class` method we're calling on it in the matcher. The fix for this is to use something like `reponds_to?(:object_class)` before calling the method. I believe this has been addressed in the serializer matcher.

*Model*:
```ruby
class Review < ApplicationRecord
  serialize :status, StatusSerializer
  validates :status, presence: true
end
```

*Serializer*:
```ruby
module StatusSerializer
  module_function

  def dump(content)
    #...
  end

  def load(content)
    #...
  end
end
```

*Test*:
```ruby
require 'rails_helper'

RSpec.describe Review, type: :model do
  it { should validate_presence_of(:status) }
end
```

*Error*:
```
 1) Review should validate that :status cannot be empty/falsy
     Failure/Error: it { should validate_presence_of(:status) }

     NoMethodError:
       undefined method `object_class' for StatusSerializer:Module
     # ./spec/models/review_spec.rb:8:in `block (2 levels) in <top (required)>'
```

*Cause*:
https://github.com/thoughtbot/shoulda-matchers/blob/fb6ea7afd4bf625edd33603c11ef4af90e914006/lib/shoulda/matchers/active_model/validate_presence_of_matcher.rb#L318

**Comment by mcmire** [comment]
Hey @theycallmeswift! We've already got this reported in #1234 but I appreciate the example. We will have this fixed shortly!

**Comment by mcmire** [comment]
@theycallmeswift I have a branch that should fix this issue. Can you temporarily switch to the `fix-presence-with-custom-serializer` branch in your Gemfile and see if that makes your test pass?

**Comment by theycallmeswift** [comment]
@mcmire yep, seems to have done the trick!

**Comment by mcmire** [comment]
This should now be fixed in 4.1.2!
